### ISSUE TITLE
Refactor common code across providers

### ISSUE THREAD

**Issue by jbw976** [crossplaneio/crossplane#83]
Across the 3 currently supported cloud providers, there is some common (duplicated) code that could be refactored to single and reusable implementations.  We should take a pass through the providers and refactor to remove this duplication.

**Comment by jbw976** [comment]
The `ProviderID` field that almost all types have in their Status is one example that could be refactored, as noted by @negz in https://github.com/crossplaneio/crossplane/pull/317#discussion_r257360982
```
// the external ID to identify this resource in the cloud provider
ProviderID string `json:"providerID,omitempty"`
```

**Comment by negz** [comment]
Let's be careful when we address this that we don't go too far with refactoring purely for the sake of deduplication. https://github.com/crossplaneio/crossplane/issues/448 contains a few links explaining this concept in more detail.

**Comment by prasek** [comment]
@jbw976 is this handled with #612 or separate?

**Comment by jbw976** [comment]
Yes, I think this is actually taken care of in the #612 epic.  I've included issues #269 and #584 in that epic to properly track those items.
